### PR TITLE
Fix some clang-tidy complaints

### PR DESCRIPTION
**Pull request by masterleinad**
Looking at https://github.com/kokkos/kokkos/pull/7430#discussion_r1800001727, I noticed that I am still seeing `clang-tidy` complaints locally. This pull request fixes those when compiling with `HPX`, `Threads`, `OpenMP` and `Serial`.

### CODE REVIEW

**Review by tpadioleau — APPROVED** [review]
LGTM!

**Comment by tpadioleau** [comment]
Do you plan to continue working on it ? Running `grep -R "std::is_[a-Z_]*[^v]<"` I could catch more of them

**Comment by dalg24** [comment]
Did you consider adding clang-tidy checks to a few more builds to increase "coverage"?

**Comment by masterleinad** [comment]
> Do you plan to continue working on it ? Running `grep -R "std::is_[a-Z_]*[^v]<"` I could catch more of them

No, I don't intend to continue looking manually. Those are all instances I could find with local `clang-tidy` configurations.

**Comment by masterleinad** [comment]
> Did you consider adding clang-tidy checks to a few more builds to increase "coverage"?

I'm not opposed to doing that in a follow-up but didn't intend to add more here.

**Comment by dalg24** [comment]
I am not blocking but won't approve either if no build is added to enforce.

**Comment by tpadioleau** [comment]
Do we understand why clang-tidy did not catch issues from the file `algorithms/src/sorting/Kokkos_BinOpsPublicAPI.hpp` before ? Isn't it used by the `Threads` backend used in the CI ?

**Comment by masterleinad** [comment]
> Do we understand why clang-tidy did not catch issues from the file `algorithms/src/sorting/Kokkos_BinOpsPublicAPI.hpp` before ? Isn't it used by the `Threads` backend used in the CI ?

You can find the `clang-tidy` results running all builds in `continuous-integration-linux.yml` in https://github.com/masterleinad/kokkos/actions/runs/11937162091/job/33272485805.

File: .github/workflows/continuous-integration-linux.yml [hunk 1]
```diff
@@ -19,7 +19,6 @@ jobs:
         cxx_extra_flags: ['']
         cmake_build_type: ['Release', 'Debug']
         backend: ['OPENMP']
-        clang-tidy: ['']
```
Comment (tpadioleau) [inline .github/workflows/continuous-integration-linux.yml:22]: Do you think we should exclude `g++` builds ? I don't know how reliable is the support compilers other than `clang++`.

Comment (masterleinad) [reply .github/workflows/continuous-integration-linux.yml:22]: It seems to work now after removing `-Wclobbered` and I think we reevaluate when problems appear.

**Comment by masterleinad** [comment]
> Can I also add a test for HPX ?

Sure, go ahead.

File: .github/workflows/continuous-integration-linux.yml (hunk 1 above)
Comment (masterleinad) [reply .github/workflows/continuous-integration-linux.yml:22]: `-Wclobbered` is included in `-Wextra` and also not really relevant for `Kokkos` AFAICT.

Comment (masterleinad) [reply .github/workflows/continuous-integration-linux.yml:22]: I don't have a strong opinion about disabling for `g++`. It's not clear to me if the build time is significantly longer and it makes writing the CI file more difficult.

**Review by cwpearson — APPROVED** [review]

**Pull request merged**